Messenger, next version

skye17 opened the pull request:
Новая версия(первая версия к сдаче 24.11) с реализацией базы данных и nio. 
Работает и обычная версия(Threaded Client и MainServer) и nio(NioMainServer и NioClient), но по nio наверное будут замечания )). 
Юнит тестов к протоколу и команде Login пока нет, а также с помощью maven проект пока собрать не удается. 
Если это возможно, то хотелось бы сдать задачу раньше 24 ноября, поэтому если есть возможность, посмотрите, пожалуйста. 
Спасибо.

Commit:
Message Service useless methods deleted

Commit:
Client-Server messenger, version 1

Commit:
Client-Server messenger, version 1

Commit:
Client-Server messenger, version 1

Commit:
Версия 2, к сдаче 10 ноября, возможно не окончательная. Базы данных пока нет, поэтому записи не сохраняются.

Commit:
Version 3

Commit:
CodeReview

arhangeldim commented:
Откоментил. Проделано много работы, про БД я думаю мы с вами еще поговорим на РК. Задание на 15 баллов. Поправьте к РК те небольшие замечания, что я отметил. Можно смерджиьт в мастер

Commit:
Small bugs fixed, problem with Dao is still unfixed..

skye17 commented:
Поправила мелкие проблемы, насчет базы и проблемы с колонками и общей реализацией еще не придумала, как исправить. Чтобы получить 20 баллов нужно исправить работу с базой, написать юнит-тесты к протоколу и команде Login, собирать проект с помощью Maven скрипта и что еще?

Спасибо.

Commit:
Problems with DAO fixed, connection pool and protocol tests are on their way..

skye17 commented:
Обновила работу с DAO объектами, теперь реализация вроде бы нигде не вылезает, осталось сделать Connection Pool в работе с бд и тесты к протоколу.

Commit:
Connection pool and many singletons

Commit:
Deleting some old files

Commit:
JsonProtocol Test

Commit:
Style

Commit:
RK1 remarks are fixed:
1) Removed redundant JSON protocol
2) Instead of just notifications new messages themselves are sent to online chat users
